discovery-provider/plugins/verified-uploads/src/handlers/users.js

```diff
@@ -1,60 +1,64 @@
-import { dp_db } from "../db.js";
-import { slack } from "../slack.js";
-import dotenv from "dotenv";
-import axios from "axios";
+import { dp_db } from '../db.js'
+import { slack } from '../slack.js'
+import dotenv from 'dotenv'
+import axios from 'axios'
+import { getPreviousState } from './utils.js'
 
-dotenv.config();
-const { IDENTITY_URL, USERS_SLACK_CHANNEL } = process.env;
+dotenv.config()
+const { IDENTITY_URL, USERS_SLACK_CHANNEL } = process.env
 const social_handle_url = (handle) =>
-  `${IDENTITY_URL}/social_handles?handle=${handle}`;
-
-export default async ({ user_id }) => {
-  const result = await dp_db("users")
-    .select("handle", "is_verified")
-    .where("user_id", "=", user_id)
-    .whereIn("is_current", ["true", "false"])
-    .orderBy("blocknumber", "desc")
-    .limit(2)
-    .catch(console.error);
-  if (result.length == 2) {
-    const current = result[0];
-    const old = result[1];
-
-    if (current.is_verified !== old.is_verified) {
-      const is_verified = current.is_verified;
-      const handle = current.handle;
-
-      // GET https://identityservice.staging.audius.co/social_handles?handle=totallynotalec
-      const { data } = await axios
-        .get(social_handle_url(handle))
-        .catch(console.error);
-
-      const { twitterVerified, instagramVerified, tikTokVerified } = data;
-
-      let source = "unknown";
-      if (twitterVerified) {
-        source = "twitter";
-      }
-      if (instagramVerified) {
-        source = "instagram";
-      }
-      if (tikTokVerified) {
-        source = "tiktok";
-      }
-
-      const header = `User *${handle}* ${
-        is_verified ? "is now" : "is no longer"
-      } verified via ${source}!`;
-
-      const body = {
-        userId: user_id,
-        handle,
-        link: `https://audius.co/${handle}`,
-        source,
-      };
-      await slack
-        .sendMsg(USERS_SLACK_CHANNEL, header, body)
-        .catch(console.error);
+  `${IDENTITY_URL}/social_handles?handle=${handle}`
+
+// TODO: send blocknumber through pg trigger
+export default async ({ user_id, blocknumber }) => {
+  const current = await dp_db('users')
+    .select('handle', 'is_verified') // get this block number
```

Comment:
might not be an issue but i don't think we're guaranteed that the blocknumber from the trigger is associated with the current user record. there could be a lag in the trigger. maybe we should just pass handle and is_verified along w the trigger? this might also impact tracks too.

might be a super rare case but i'm thinking something like 
user becomes verified and immediately does another update.
handler is called but at this point the current / previous block state are both verified

Reply:
Tracks won't have this issue as it doesn't rely on block number. It pulls the timestamps from the trigger itself and compares them. If they match then it was an upload.

>handler is called but at this point the current / previous block state are both verified
This should still result in a comparison where one of the blocks in the reverts table still contains a `is_verified` is false. Even if the current record is farther ahead we're still reading all the updates in order. The trigger sends the blocknumber that the update occurred in, not the current block when the handler consumes the message.

Reply:
ah truee should be ok